hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/repository/NftAllowanceRepository.java

```diff
@@ -18,6 +18,29 @@
 
 import com.hedera.mirror.common.domain.entity.AbstractNftAllowance.Id;
 import com.hedera.mirror.common.domain.entity.NftAllowance;
+import java.util.List;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 
-public interface NftAllowanceRepository extends CrudRepository<NftAllowance, Id> {}
+public interface NftAllowanceRepository extends CrudRepository<NftAllowance, Id> {
+
+    /**
+     * This repository method will query based on the spender and further filter by the owner and token_id, order by owner first and then the token_id
+     * 1. It only covers owner eq and token_id gte scenario.
+     */
+    @Query(
+            value =
+                    "select * from nft_allowance where spender = ?  and  owner = ? and token_id >= ? order by owner,token_id limit ?",
```

Comment:
Probably this is the most common static query to have. If they omit owner or token_id you can default them to 0. If they omit direction you can default to ASC. If they omit limit you can default to 25. If they supply an `account.id=gte:X` you can subtract 1 from X. Similar for `token.id=gte:Y`.

```suggestion
                    "select * from nft_allowance where spender = ? and owner > ? and token_id > ? order by owner ?, token_id ? limit ?",
```